framework/aster-frame/src/trap/handler.rs

```diff
@@ -6,23 +6,23 @@ use trapframe::TrapFrame;
 
 use crate::{arch::irq::IRQ_LIST, cpu::CpuException, cpu_local};
 
-pub(crate) fn call_irq_callback_functions(trap_frame: &TrapFrame) {
+pub(crate) fn call_irq_callback_functions(trap_frame: &TrapFrame, irq_number: usize) {
     // For x86 CPUs, interrupts are not re-entrant. Local interrupts will be disabled when
     // an interrupt handler is called (Unless interrupts are re-enabled in an interrupt handler).
     //
     // FIXME: For arch that supports re-entrant interrupts, we may need to record nested level here.
     IN_INTERRUPT_CONTEXT.store(true, Ordering::Release);
 
-    let irq_line = IRQ_LIST.get().unwrap().get(trap_frame.trap_num).unwrap();
+    let irq_line = IRQ_LIST.get().unwrap().get(irq_number).unwrap();
     let callback_functions = irq_line.callback_list();
     for callback_function in callback_functions.iter() {
         callback_function.call(trap_frame);
     }
     drop(callback_functions);
 
-    if !CpuException::is_cpu_exception(trap_frame.trap_num as u16) {
-        crate::arch::interrupts_ack();
-    }
+    // if !CpuException::is_cpu_exception(irq_number as u16) {
+    //     crate::arch::interrupts_ack();
+    // }
```

Comment:
Would it be better to move this "cleanup" code to `arch`? Intuitively it's part of exception handling instead of the IRQ callback. Although RISC-V also needs a ACK, but it takes a `irq_number`-like argument.